runtime/src/accounts_db.rs

```diff
@@ -396,20 +396,15 @@ impl<'a, 'b> Serialize for AccountsDBSerialize<'a, 'b> {
 
 #[derive(Clone, Default, Debug, Serialize, Deserialize, PartialEq)]
 pub struct BankHashStats {
-    pub num_removed_accounts: u64,
-    pub num_added_accounts: u64,
+    pub num_accounts: u64,
     pub num_lamports_stored: u64,
     pub total_data_len: u64,
     pub num_executable_accounts: u64,
 }
 
 impl BankHashStats {
     pub fn update(&mut self, account: &Account) {
-        if Hash::default() == account.hash {
```

Comment:
@jstarry well, this is very useful when I'm debugging AccountsDB mysteries. Could you make this `fn` take computed `Hash` as an argument somehow so that we can still distinguish between `add` and `removed` as-is?

Reply:
@ryoqun sure! Why is a non default hash considered a "removed" account?

Reply:
Well, yeah, that's a bit misleading wording. From user's perspective, it could be better described as `updated`. This `remove` means _removing_ the `account.hash` from no-longer-existing Xor hash.

@sakridge WDYT about renaming from `num_removed_accounts` to `num_updated_accounts`?

Reply:
Threading `Hash` through is a bit of a pain. We store accounts in so many different ways 😵 If the account update stat is important, then maybe we should leave `account.hash` as is for now? I don't think this change is worth the effort right now.

For context, I was concerned that we don't check if `account.hash` was tampered with after message processing. But since `hash` is only currently used for stats, it's pretty harmless.

Reply:
@jstarry Super thanks for giving it a shot! I'm really sorry. I have one thing to apologize... Come to think of it, there is no need to thread `Hash` through albeit I said so. The information of `Hash:default() == account.hash` can easily derived from just `account.lamports == 0` (see `hash_account_data`).

Reply:
So, I guess we can remove `account.hash` without any loss of logging detail and any downside whatsoever. :)

Reply:
@ryoqun Ahhh, I was confused before. I thought we were trying to track which accounts will be newly _added_ to the hash. But `lamports == 0` allows us to check which accounts will be _removed_. Is it not needed to track which accounts were added?

Reply:
> @ryoqun I thought we were trying to track which accounts will be newly _added_ to the hash. But `lamports == 0` allows us to check which accounts will be _removed_. Is it not needed to track which accounts were added?

Oops. sorry again. I was wrong again. You're right. Yeah, confusing `account.hash`....

I think the new metrics (_the number of accounts to be updated vs removed_) is enough to be a replacement to remove `account.hash`. @sakridge Wdyt? In this pr, @jstarry and I are trying to remove `account.hash`, which seems not to be needed. The only remaining minor (imo) downside is that this removed metrics isn't easily computable anymore. As for merits, we can shave off 32 bytes per account when we're juggling inside the runtime, that might be worthwhile. :)

Reply:
Yea, it's not a huge loss. I'm ok with removing it.